### PR TITLE
Save to GitHub

### PR DESCRIPTION
**Pull request by shahronak47**
Hi Andres, 

This PR aims to create a new function called `save_to_gh` which aims to save the data to GitHub. This is written mostly to save GDP  data to GitHub for this task. https://app.clickup.com/t/8688xw9p1

### CODE REVIEW

File: R/save_to_gh.R [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    repo    = repo,
+    path    = glue::glue("{filename}.{ext}"),
+    .params = list(
+      branch  = branch,
+      message = "updating data",
+      sha     = out$sha,
+      content = pipaux::convert_df_to_base64(df)
+    ),
+    .token = Sys.getenv('GITHUB_PAT')
+  )
```
Comment (randrescastaneda) [inline R/save_to_gh.R]: Hi @shahronak47 , 

Thanks. Make sure you use the `.token` argument as we do here: https://github.com/PIP-Technical-Team/pipfun/blob/e9a53060c396c4ca6e090f15b57a26f91e82084d/R/load_from_gh.R#L108

File: tests/testthat/test-save_to_gh.R [hunk 2]
```diff
@@ -0,0 +1,5 @@
+test_that("save_to_gh works correctly", {
+  Sys.setenv(GITHUB_PAT = 'code')
+  mockery::stub(save_to_gh, "gh::gh", NULL)
```
Comment (randrescastaneda) [inline tests/testthat/test-save_to_gh.R]: Hi @shahronak47 
The `mockery` package is being replaced by the new mocking tools in testthat: https://testthat.r-lib.org/reference/local_mocked_bindings.html

You can take a look at how we use them in the `pipapi` for some testing
1. For specific calls: https://github.com/PIP-Technical-Team/pipapi/blob/c1267780f08af67e18c0023ecfcfae887e203587/tests/testthat/test-ui_poverty_indicators.R#L18-L27
2. For complete tests: https://github.com/PIP-Technical-Team/pipapi/blob/c1267780f08af67e18c0023ecfcfae887e203587/tests/testthat/test-pip_grp_logic.R#L27-L29 


Of course, if these `testhat` functions do not work in your case, please keep using `mockery`. 😉 
Thanks.

**Review by randrescastaneda — CHANGES_REQUESTED** [review]
HI @shahronak47 , 

Thank you so much. I added a couple of minor comments. 

Thank you so much. 

Best,

File: R/save_to_gh.R (hunk 1 above)
Comment (shahronak47) [reply R/save_to_gh.R]: Done.

File: tests/testthat/test-save_to_gh.R (hunk 2 above)
Comment (shahronak47) [reply tests/testthat/test-save_to_gh.R]: Thanks, I didn't know `mockery` was superseded. Replaced `mockery::stub` with `local_mocked_bindings`

File: R/save_to_gh.R [hunk 3]
```diff
@@ -37,10 +39,11 @@ save_to_gh <- function(df,
       branch  = branch,
       message = "updating data",
       sha     = out$sha,
-      content = pipaux::convert_df_to_base64(df)
+      content = pipaux:::convert_df_to_base64(df)
```
Comment (randrescastaneda) [inline R/save_to_gh.R]: hi @shahronak47 , 

I think we have a circular dependency problem here. `pipaux` depends on `pipfun` (see [here](https://github.com/PIP-Technical-Team/pipaux/blob/88429b29adc17d02d6feae15784fec73dfe2b25a/DESCRIPTION#L51)). 

It think the best option is to move `convert_dr_to_base64()` to `pipfun` from `pipaux`.  What do you think?

**Review by randrescastaneda — CHANGES_REQUESTED** [review]
Hi @shahronak47 , 

please take a look at the circular dependency problem we have with `pipfun`. 

Thanks.

File: R/save_to_gh.R (hunk 3 above)
Comment (shahronak47) [reply R/save_to_gh.R]: This makes sense, I have added an update to copy the function here. Once this is merged, will remove the function from `pipaux`.

**Review by randrescastaneda — APPROVED** [review]
Fantastic

**Pull request merged**